tests/test_cli_get_set.py

```diff
@@ -321,25 +454,25 @@ def test_set_override_multiple_existing_files(self):
         file2 = os.path.join(self.workdir.name, 'etc', 'netplan', 'eth1.yaml')
         with open(file2, 'w') as f:
             f.write(r'network: {ethernets: {eth1: {dhcp4: true}}}')
-        self._set([(r'network={renderer: NetworkManager, version: 2,'
-                    r'ethernets:{'
-                    r'eth1:{dhcp4: false},'
-                    r'eth0.1:{dhcp4: false},'
-                    r'eth0.2:{dhcp4: false}},'
-                    r'bridges:{'
-                    r'br99:{dhcp4: false}}}')])
+        self._set([(r'network={renderer: NetworkManager, version: 2, '
+                    r'ethernets: {'
+                    r'eth1: {dhcp4: false},'
+                    r'eth0.1: {dhcp4: false},'
+                    r'eth0.2: {dhcp4: false}},'
+                    r'bridges: {'
+                    r'br99: {dhcp4: false}}}')])
```

Comment:
question: Is this just some cleanup or an actual change in behavior, that we now require properly space separated mappings? We would need to take care not to break any consumer (like our own `netplan-dbus` daemon and/or `snapd`'s get/set functionality).

Reply:
Oh crap. It seems it is a change in behavior, I get `netplan.libnetplan.LibNetplanException: Error parsing YAML: found unexpected ':'` when reverting the changes.
This seems weird to me, as the underlying parsing library is the same, isn't it?

FWIW, I checked the YAML spec, and at least in 1.1 and 1.2 the whitespace after the colon is mandatory.

Reply:
I don't know for sure if PyYAML is using libyaml in the background, too. If those two YAML implementations differ that would be a problem, especially as we have consumers using that (out of spec?) syntax: https://github.com/snapcore/snapd/pull/10752/files#diff-e380767e9d06e93d76ea779341439e7d4c878b2708bcc23731453c17c2442616R557

I wonder if we can somehow handle both cases and maybe print a warning that this does not comply with the YAML specification?

Reply:
After investigations, I note the following things:

* Even though PyYAML loads libyaml, it does *not* use it in the default case. You have to explicitly use the CLoader class in order to do so. By default it uses a pure Python parser, which is apparently less strict than the C one.
* When using CLoader on the original string, I get the same error.
* However, when using the one in the snapd source code, it loads just fine. The difference is that the snapd code quotes their keys, since they're using a JSON serializer to format their string (which is valid since JSON is a subset of YAML). So, on that front I think we're OK.

Reply:
To close up this thread: I've added a new commit that explicitly switches this to a JSON-payload to match the snapd use-case.